google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/BigtableBatchingCallSettings.java

```
@@ -145,9 +193,106 @@ public Builder setRetrySettings(@Nonnull RetrySettings retrySettings) {
+     *
+     * <p>If latency based throttling is enabled, initialOutstandingElementCount = Math.max(batch
+     * element count, maxOutstandingElementCount / 4); minOutstandingElementCount = Math.max(batch
+     * element count, maxOutstandingElementCount / 100). If latency based throttling is disabled,
+     * initialOutstandingElementCount, maxOutstandingElementCount and minOutstandingElementCount
+     * will be the same.
+     *
+     * <p>Latency based throttling only adjusts outstanding element count.
+     * initialOutstandingRequestBytes, minOutstandingRequestBytes and maxOutstandingRequestBytes
+     * will always be the same.
+     *
+     * <p>{@link LimitExceededBehavior} will always be {@link LimitExceededBehavior#Block}.
+     */
+    public DynamicFlowControlSettings getDynamicFlowControlSettings() {
+      return this.dynamicFlowControlSettings;
+    }
+
     /** Builds the {@link BigtableBatchingCallSettings} object with provided configuration. */
     @Override
     public BigtableBatchingCallSettings build() {
+      Preconditions.checkState(batchingSettings != null, "batchingSettings must be set");
+      FlowControlSettings defaultSettings = batchingSettings.getFlowControlSettings();
+      if (isLatencyBasedThrottlingEnabled()) {
+        Long maxThrottlingElementCount = defaultSettings.getMaxOutstandingElementCount();
+        Long maxThrottlingRequestByteCount = defaultSettings.getMaxOutstandingRequestBytes();
+        if (maxThrottlingElementCount == null) {
+          long maxBulkMutateElementPerBatch = 100L;
+          long defaultChannelPoolSize = 2 * Runtime.getRuntime().availableProcessors();
+          maxThrottlingElementCount =
+              Math.min(20_000L, 10L * maxBulkMutateElementPerBatch * defaultChannelPoolSize);
+        }
+        if (maxThrottlingRequestByteCount == null) {
+          maxThrottlingRequestByteCount = 100L * 1024 * 1024;
+        }
```

It seems like you are duplicating the defaults from EnhancedBigtableStubSettings. This will be very hard to keep in sync. If these settings are required and the user hasnt set them, then I think we should raise an error.